src/server/services/Actions.ts

```diff
@@ -164,15 +178,20 @@ export default class Actions {
   ): Promise<DripResponse> {
     let dripTimeout: ReturnType<typeof rpcTimeout> | null = null;
     let result: DripResponse;
+    const parsedAmount = Number(amount);
+    const faucetBalance = this.getFaucetBalance();
 
     try {
       if (!this.account) throw new Error('account not ready');
 
-      const dripAmount = Number(amount) * 10 ** decimals;
-
+      if (faucetBalance && parsedAmount >= faucetBalance) {
+        throw new Error(
+          `Can't send "${parsedAmount}", as balance is smaller "${faucetBalance}"`
+        );
+      }
       // start a counter and log a timeout error if we didn't get an answer in time
```

Comment:
This comment should be moved above `dripTimeout` I presume

Reply:
[c513bbb](https://github.com/paritytech/substrate-matrix-faucet/pull/147/commits/c513bbba76ffeddfc0a5022380e7179fc87e2d97)